packages/router/src/Set.tsx

```diff
@@ -30,7 +30,7 @@ type SetProps<P> = P & {
   prerender?: boolean
   children: ReactNode
   /** Loading state for auth to distinguish with whileLoading */
-  whileLoadingAuth?: () => React.ReactElement | null
+  whileLoadingAuth?: () => React.ReactElement
```

Comment:
The union with `null` seemed unnecessary since we're typing it as optional already.

Reply:
Optional here means `whileLoadingAuth` is optional, i.e. you don't have to include it as a prop on `<Set>`.
`| null` means `whileLoadingAuth` is allowed to return `null`.

Reply:
Ah I see... so it's not a union type, but a union of the function's return type right?

Reply:
Exactly :)